common/flatpak-transaction.c

```diff
@@ -2843,7 +2854,116 @@ flatpak_transaction_add_rebase (FlatpakTransaction *self,
   if (dir_ref_is_installed (priv->dir, decomposed, &installed_origin, NULL))
     remote = installed_origin;
 
-  return flatpak_transaction_add_ref (self, remote, decomposed, subpaths, previous_ids, NULL, FLATPAK_TRANSACTION_OPERATION_INSTALL_OR_UPDATE, NULL, NULL, FALSE, error);
+  return flatpak_transaction_add_ref (self, remote, decomposed, subpaths, previous_ids, NULL, FLATPAK_TRANSACTION_OPERATION_INSTALL_OR_UPDATE, NULL, NULL, FALSE, NULL, error);
+}
+
+/**
+ * flatpak_transaction_add_rebase_and_uninstall:
+ * @self: a #FlatpakTransaction
+ * @remote: the name of the remote
+ * @old_ref: the ref to uninstall
+ * @new_ref: the ref to rebase to
```

Comment:
The function name puts the verb `rebase` before `uninstall`, so I think it might be more natural to put the new ref before the old:

```
flatpak_transaction_add_rebase_and_uninstall (txn, remote,
                                              "app/net.example.NewName/x86_64/stable",
                                              "app/org.example.OldName/x86_64/stable", ...)
```

Reply:
Sure. I was going for ordering the arguments as old → new, but your reasoning makes sense too. Changed.